src/cs/Ssh/IO/SshProtocol.cs

```diff
@@ -160,19 +163,25 @@ internal async Task<string> ReadProtocolVersionAsync(CancellationToken cancellat
 				if (line.StartsWith("SSH-", StringComparison.Ordinal))
 				{
 					this.metrics.AddMessageReceived(i + 1);
-					return line;
+					return (line, proxyIPAddress);
+				}
+				else if (line.StartsWith("PROXY ", StringComparison.Ordinal))
```

Comment:
Are we going to authenticate the other party independently before we trust the proxy version? `ReadProtocolVersion` happens before the encryption and authentication, anyone can snip it if the SSH session runs in a non-protected connection, and a malicious client can fake it. 

In context of dev tunnels, snipping from unprotected channels is less of a concern because of https. However, we must not use this "PROXY" IP protocol version extension for authentication purposes.